sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationClientBuilder.java

```diff
@@ -0,0 +1,386 @@
+     * @param elevationClientId the clientId value.
+     * @return the ElevationClientBuilder.
+     */
+    public ElevationClientBuilder elevationClientId(String elevationClientId) {
+        this.elevationClientId = Objects.requireNonNull(elevationClientId, "'elevationClientId' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Set endpoint of the service.
+     *
+     * @param endpoint url of the service
+     * @return ElevationClientBuilder
+     */
+    @Override
+    public ElevationClientBuilder endpoint(String endpoint) {
+        this.endpoint = Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the {@link ElevationServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param version {@link ElevationServiceVersion} of the service to be used when making requests.
+     * @return the updated ElevationClientBuilder object
+     */
+    public ElevationClientBuilder serviceVersion(ElevationServiceVersion version) {
+        this.serviceVersion = version;
+        return this;
+    }
+
+    /**
+     * Sets The HTTP pipeline to send requests through.
+     *
+     * @param pipeline the pipeline value.
+     * @return the ElevationClientBuilder.
+     */
+    @Override
+    public ElevationClientBuilder pipeline(HttpPipeline pipeline) {
+        if (this.pipeline != null && pipeline == null) {
+            LOGGER.info("Pipeline is being set to 'null' when it was previously configured.");
+        }
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    /**
+     * Sets The HTTP client used to send the request.
+     *
+     * @param httpClient the httpClient value.
+     * @return the ElevationClientBuilder.
+     */
+    @Override
+    public ElevationClientBuilder httpClient(HttpClient httpClient) {
+        if (this.httpClient != null && httpClient == null) {
+            LOGGER.info("HttpClient is being set to 'null' when it was previously configured.");
+        }
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets The configuration store that is used during construction of the service client.
+     *
+     * @param configuration the configuration value.
+     * @return the ElevationClientBuilder.
+     */
+    @Override
+    public ElevationClientBuilder configuration(Configuration configuration) {
+        this.configuration = Objects.requireNonNull(configuration, "'configuration' cannot be null.");
```

Comment:
`Configuration`, `HttpLogOptions`, `RetryPolicy`, and `ClientOptions` shouldn't throw an exception and just use a default later on when the client is being built

Reply:
This will be updated in the next version of beta for consistency across modules that have been released